lib/puppet/util/logging.rb

```diff
@@ -281,8 +281,8 @@ def issue_deprecation_warning(message, key, file, line, use_caller)
         $deprecation_warnings[key] = message
         call_trace = use_caller ?
           (offender || get_deprecation_offender).join('; ') :
-          "#{file || 'unknown'}:#{line || 'unknown'}"
-        warning("#{message}\n   (at #{call_trace})")
+          "#{file || _('unknown')}:#{line || _('unknown')}"
```

Comment:
which kind of begs the question - how would 'unknown' get translated? maybe define an unknown string variable first?

ie.
unknown = _("unknown")
_("%{file}:%{line}") % { file: file || unknown, line: line || unknown }